addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -694,6 +694,8 @@ var SnippetEditor = Widget.extend({
         this.styles = {};
         this.selectorSiblings = [];
         this.selectorChildren = [];
+        this.selectorLockWithin = [];
+        const selectorExcludeParent = new Set();
```

Comment:
I suppose that you use a `Set` for `selectorExcludeParent` to prevent duplicate entries right ? Why do you do it for `selectorExcludeParent` and not for `selectorLockWithin` ?